src/ctapipe/utils/astro.py

```diff
@@ -4,63 +4,230 @@
+        Radius of the sky region around the pointing position. Default is the full sky.
+    magnitude_cut : float, optional
+        Return only stars above a given apparent magnitude. Default is None (all entries).
+    band : str, optional
+        Wavelength band to use for the magnitude cut. Options are 'Vmag' and 'B'. Default is 'Vmag'.
 
     Returns
     -------
-    Astropy table:
-       List of all stars after cuts with names, catalog numbers, magnitudes,
-       and coordinates
+    astropy.table.Table
+        List of all stars after applying the cuts, with the same keys as the input table `stars`.
     """
-    from ctapipe.utils import get_table_dataset
-
-    catalog = get_table_dataset("yale_bright_star_catalog5", role="bright star catalog")
-
-    starpositions = SkyCoord(
-        ra=Angle(catalog["RAJ2000"], unit=u.deg),
-        dec=Angle(catalog["DEJ2000"], unit=u.deg),
-        frame="icrs",
-        copy=False,
-    )
-    catalog["ra_dec"] = starpositions
-
-    if magnitude_cut is not None:
-        catalog = catalog[catalog["Vmag"] < magnitude_cut]
+    stars_ = deepcopy(stars)
+    if magnitude_cut:
+        try:
+            stars_ = stars_[stars_[band] < magnitude_cut]
+        except KeyError:
+            raise ValueError(
+                f"The requested catalogue has no compatible magnitude for the {band} band"
+            )
 
     if radius is not None:
-        if pointing is None:
+        if pointing:
+            stars_["separation"] = stars_["ra_dec"].separation(pointing)
+            stars_ = stars_[stars_["separation"] < radius]
+        else:
             raise ValueError(
                 "Sky pointing, pointing=SkyCoord(), must be "
                 "provided if radius is given."
             )
-        separations = catalog["ra_dec"].separation(pointing)
-        catalog["separation"] = separations
-        catalog = catalog[separations < radius]
 
-    catalog.remove_columns(["RAJ2000", "DEJ2000"])
+    return stars_
+
+
+def get_star_catalog(
+    catalog: str | StarCatalog, magnitude_cutoff: float = 8.0, row_limit: int = 1000000
+) -> Table:
+    """
+    Utility function to download a star catalog for the get_bright_stars function.
+
+    Parameters
+    ----------
+    catalog : str or ctapipe.utils.astro.StarCatalog
+        Name of the catalog to be used. Usable names are found in the Enum StarCatalog. Default: Yale.
+    magnitude_cutoff : float, optional
+        Maximum value for magnitude used in lookup. Default is 8.0.
+    row_limit : int, optional
+        Maximum number of rows for the star catalog lookup. Default is 1000000.
+
+    Returns
+    -------
+    astropy.table.Table
+        List of all stars after cuts with catalog numbers, magnitudes,
+        and coordinates as SkyCoord objects including proper motion.
+    """
+    from astroquery.vizier import Vizier
+
+    if isinstance(catalog, str):
+        catalog = StarCatalog[catalog]
+    catalog_dict = catalog.value
```

Comment:
not a dict anymore. Prefer better names to encoding type in variable name. E.g. `catalog_info`